File: sonet-api/internal/model/session.go
package model

import "time"

type Session struct {
	ID             uint64    `json:"id" db:"id"`
	UserID         uint64    `json:"user_id" db:"user_id"`
	RefreshTokenID *uint64   `json:"refresh_token_id" db:"refresh_token_id"`
	DeviceName     *string   `json:"device_name" db:"device_name"`
	DeviceType     string    `json:"device_type" db:"device_type"`
	OS             *string   `json:"os" db:"os"`
	Browser        *string   `json:"browser" db:"browser"`
	IPAddress      *string   `json:"ip_address" db:"ip_address"`
	Location       *string   `json:"location" db:"location"`
	LastActiveAt   time.Time `json:"last_active_at" db:"last_active_at"`
	CreatedAt      time.Time `json:"created_at" db:"created_at"`
	IsCurrent      bool      `json:"is_current" db:"is_current"`
}

File: sonet-web/src/stores/session.ts
import { defineStore } from 'pinia'
import { ref } from 'vue'
import api from '@/composables/useApi'

export interface Session {
  id: number
  user_id: number
  device_name: string | null
  device_type: string
  os: string | null
  browser: string | null
  ip_address: string | null
  location: string | null
  last_active_at: string
  created_at: string
  is_current: boolean
}

export const useSessionStore = defineStore('session', () => {
  const sessions = ref<Session[]>([])
  const loading = ref(false)

  async function fetchSessions() {
    loading.value = true
    try {
      const { data } = await api.get('/sessions')
      sessions.value = data.data || []
    } catch {
      sessions.value = []
    } finally {
      loading.value = false
    }
  }

  async function revokeSession(sessionId: number) {
    await api.delete(`/sessions/${sessionId}`)
    sessions.value = sessions.value.filter((s) => s.id !== sessionId)
  }

  async function revokeAllOther(currentSessionId: number) {
    await api.post('/sessions/revoke-all', { current_session_id: currentSessionId })
    sessions.value = sessions.value.filter((s) => s.id === currentSessionId)
  }

  return {
    sessions,
    loading,
    fetchSessions,
    revokeSession,
    revokeAllOther,
  }
})
